### PR TITLE
Improve capitalization of font names in method names

### LINKED ISSUE #87: Improperly capitalized font names
The current transformation from font names into Dart identifiers sometimes chooses an ugly capitalization. For example, "IBM Plex Mono" becomes `GoogleFonts.iBMPlexMono` and "IM Fell DW Pica SC" becomes `GoogleFonts.iMFellDWPicaSC`.

Let's see what the [Effective Dart](https://dart.dev/guides/language/effective-dart) guidelines have to say about this topic! There are two somewhat relevant rules:
- [**DO name other identifiers using `lowerCamelCase`.**](https://dart.dev/guides/language/effective-dart/style#do-name-other-identifiers-using-lowercamelcase)
  Class members, top-level definitions, variables, parameters, and named parameters should capitalize the first letter of each word except the first word, and use no separators.
- [**DO capitalize acronyms and abbreviations longer than two letters like words.**](https://dart.dev/guides/language/effective-dart/style#do-capitalize-acronyms-and-abbreviations-longer-than-two-letters-like-words)
  Capitalized acronyms can be hard to read, and multiple adjacent acronyms can lead to ambiguous names. For example, given a name that starts with `HTTPSFTP`, there’s no way to tell if it’s referring to HTTPS FTP or HTTP SFTP.
  To avoid this, acronyms and abbreviations are capitalized like regular words, except for two-letter acronyms. (Two-letter _abbreviations_ like ID and Mr. are still capitalized like words.)
  **Examples** include `HttpRequest` instead of `HTTPRequest` and `Id` instead of `ID`.

So, I believe we should only capitalize the first letter of each word of the font name (except the first one). "IBM Plex Mono" would become `GoogleFonts.ibmPlexMono` and "IM Fell DW Pica SC" would become `GoogleFonts.imFellDwPicaSc`.

I'd be happy to hear your opinion and implement the necessary changes.

### PR DESCRIPTION
**Pull request by marcelgarus**
Fixes #87 😊
Because this breaks the code of users that used a font that got another name, this is a breaking change and should cause a version bump to 0.5.0.

### CODE REVIEW

File: generator/generator.dart [hunk 1]
```diff
@@ -92,6 +91,18 @@ void _generateDartFile(Directory fontDirectory) {
   outFileWriteSink.close();
 }
 
+String _familyToMethodName(String family) {
+  final words = family.split(' ');
+  for (var i = 0; i < words.length; i++) {
+    final word = words[i];
+    final isFirst = (i == 0);
```
Comment (johnsonmh) [inline generator/generator.dart:98]: optional/nit: you don't need the parentheses on this line and the one below it.

**Comment by marcelgarus** [comment]
Done 😊 You'll have to merge this, I don't have write access to this repo.

**Comment by Zazo032** [comment]
Hi! I think this version is missing on `CHANGELOG.md`. As it seems to be a breaking change, would it be possible to add it?

**Comment by johnsonmh** [comment]
@Zazo032 Yes good catch, changelog is now updated on pub.dev.

**Pull request merged**